app/controllers/chat_channels_controller.rb

```diff
@@ -184,6 +184,38 @@ def search
     }, ChatChannelSearchSerializer, root: false)
   end
 
+  def archive
+    params.require(:chat_channel_id)
+    params.require(:type)
+
+    topic_type = params[:type] == "newTopic" ? :new : :existing
+    topic_params_ok = \
+      (topic_type == :existing && params[:topic_id].present?) ||
+      (topic_type == :new && params[:title].present?)
```

Comment:
that topic_type is not used further down in the method, so I don't see the benefit of that assignment

```
    topic_params_ok = params[:type] == "newTopic" ? params[:title].present? : params[:topic_id].present?
```
or maybe even 
```
if params[:type] == "newTopic" ? params[:title].present? : params[:topic_id].present?
  raise Discourse::InvalidParameters 
end
```

Reply:
Good point, I think I planned to use that type at some point but didn't